File: rootmanager/src/main/java/com/creepersan/rootmanager/RootManager.java
package com.creepersan.rootmanager;

import java.io.IOException;

public class RootManager {
    private static volatile RootManager mInstance = null;

    public static RootManager getInstance(){
        if (mInstance == null){
            synchronized (RootManager.class){
                if (mInstance == null){
                    mInstance = new RootManager();
                }
            }
        }
        return mInstance;
    }

    private RootManager(){

    }

    public boolean checkHasRootPermission(){
        Runtime runtime = Runtime.getRuntime();
        try {
            Process process = runtime.exec("su");
            int resultCode = process.waitFor();
            return resultCode == 0;
        } catch (InterruptedException e) {
            e.printStackTrace();
            return false;
        } catch (IOException e) {
            e.printStackTrace();
        }
        return true;
    }

    public Conversation createConversation(){
        return new Conversation(Runtime.getRuntime());
    }


}

File: root_manager/src/main/java/com/creepersan/root_manager/RootManager.kt
package com.creepersan.root_manager

import com.creepersan.root_manager.command.Command
import com.creepersan.root_manager.command.CommandCallback
import com.creepersan.root_manager.terminal.Terminal

class RootManager private constructor(){

    companion object{
        @Volatile private var mInstance : RootManager? = null

        fun getInstance(): RootManager {
            if (mInstance == null){
                synchronized(this){
                    if (mInstance == null){
                        mInstance = RootManager()
                    }
                }
            }
            return mInstance!!
        }
    }

    private val mTerminal = Terminal(Runtime.getRuntime())

    fun checkHasRootPermission(callback:CommandCallback){
        mTerminal.execute(Command("su", callback))
    }

    fun sendCommand(command: Command){
        mTerminal.execute(command)
    }

    fun action(){
        mTerminal.action()
    }

}